### DISCUSSION TITLE
Enable/Disable Use Timer Sensor

### DISCUSSION THREAD

**Discussion by francoisoli4** [kellerza/sunsynk#137] [General]
Hi @kellerza 

Can you please advise if there is a sensor one can use to enable/disable the "use timer" option under work mode 1?

I would like to use work mode 2 during the day and work mode 1 at night, and would like to have the timer enabled/disabled using an automation.

![Capture](https://user-images.githubusercontent.com/83315275/237042652-d39e4d39-89cc-43ca-be57-6b97508c0b16.JPG)

Thank you
(ad thank you for the great work with this integration)

**Comment by kellerza** [comment]
Hi,
It might be register 248? You can try the following custom sensor: 

SENSORS += SwitchRWSensor(248, "Use Timer", on=255)

Let me know if this works...

You can get the list of registers from #59

**Reply by francoisoli4** [reply]
Hi @kellerza 

Thanks a lot for the swift reply. I did what you said, but the addon crashes after adding the custom sensor with a "Name Error".
Please see log below:

![image](https://github.com/kellerza/sunsynk/assets/83315275/4eeb0ac5-e5d7-4f95-871d-41ff922010a4)

**Reply by kellerza** [reply]
Ensure that you import SwitchRWSensor, it's probably not imported yet... (there might already be other classes imported from that module in which case you can add it to the list)

```python
from sunsynk.rwsensors import SwitchRWSensor
```

**Reply by francoisoli4** [reply]
Thank you great Sir. That did the job. Will I be able to use this as a switch to disable/enable it from within HA?

*sorry for all the noob questions

**Reply by francoisoli4** [reply]
When the "Use timer" box is ticked manually on the inverter I get the following output:
![Use Timer On](https://github.com/kellerza/sunsynk/assets/83315275/91be6067-722e-4cb9-8316-09848d580d23)

When the "use timer" box is unticked manually on the inverter I get the following output with a WARNING:
![Use Timer OFF](https://github.com/kellerza/sunsynk/assets/83315275/78b7b35b-f475-4c1d-ba78-800c2372aa38)

**Reply by kellerza** [reply]
In that case I would add bitmask=1 and on=1 (instead of 255 and no bitmask)

```python
SwitchRWSensor(248, "Use Timer", on=1, bitmask=1)
```

**Comment by hiltctn** [comment]
@francoisoli4   Did you ever get this right ?  If so, could you explain how ?  I'm looking to automate something similar

**Reply by francoisoli4** [reply]
Are you talking about the "use Timer" as a switch in HA?

**Reply by hiltctn** [reply]
Yes sir.    I'd like HA to toggle the "use Timer" like an on/off switch

**Comment by francoisoli4** [comment]
you have to add the "use_timer" sensor in the config.
![image](https://github.com/kellerza/sunsynk/assets/83315275/eebbe2c2-bc1e-4b10-87ca-52d2a04ebbb0)

I then added a template switch in my configuration.yaml file (below) to be able to switch the timer on/off. Works like a charm.
![image](https://github.com/kellerza/sunsynk/assets/83315275/0158a335-1446-4a4c-9a4a-9d993e4f53d9)

Hope this makes sense

**Reply by hiltctn** [reply]
Ah.  Fabulous !  This looks great !  I'm only 3 weeks into the rabbit hole that is HA.  Haven't gotten to the templating bit yet.  But this looks like the answer to my question.  I'll give it a go soonest

My use case :  My battery seldom gets up to 100% during the winter months.  And at 100% is when it performs its self maintenance tasks like cell balancing etc.   So ideally, once or twice a month I'd like HA to run a task which turns off the inverter timer so that the battery can charge up, balance etc.  Then a few hours later it can switch said timer back on and all is normal

Thanks again for this awesomeness

**Reply by PJH-ZA** [reply]
I have added the use_timer sensor as above. Please can you explain exactly where I need to go to complete the following : I then added a template switch in my configuration.yaml file to be able to switch the timer on/off. Still new to HA so not sure under which section or how to add this toggle switch. Please can you explain. Thank you

**Comment by reedy** [comment]
> I then added a template switch in my configuration.yaml file (below) to be able to switch the timer on/off. Works like a charm. ![image](https://user-images.githubusercontent.com/83315275/245795314-0158a335-1446-4a4c-9a4a-9d993e4f53d9.png)

Because we're all lazy, here's a copy-paste-able version:

```yaml
switch:
  - platform: template
    switches:
      sunsynk_use_timer_switch:
         friendly_name: "Sunsynk Use Timer"
         value_template: "{{ is_state('switch.ss_use_timer', 'on') }}"
         turn_on:
           service: switch.turn_on
           entity_id: switch.ss_use_timer
         turn_off:
           service: switch.turn_off
           entity_id: switch.ss_use_timer
         icon_template: mdi:timer
```

**Comment by PJH-ZA** [comment]
I have added the use_timer sensor as above. Please can you explain exactly where I need to go to complete the following : I then added a template switch in my configuration.yaml file to be able to switch the timer on/off. Still new to HA so not sure under which section or how to add this toggle switch. Please can you explain. Thank you

**Reply by hiltctn** [reply]
> I have added the use_timer sensor as above. Please can you explain exactly where I need to go to complete the following : I then added a template switch in my configuration.yaml file to be able to switch the timer on/off. Still new to HA so not sure under which section or how to add this toggle switch. Please can you explain. Thank you

Hi PJH-ZA

I ultimately ended up going a different route ( ESP32 with Slipx06 Sunsynk ESPHome code ).

That said, if you've managed to successfully create a template switch per above, the switch should be visible on your default HA dashboard / landing page as a switch that you can toggle.  From there, you use it as part of an automation, particularly in the Action section.   

So, for example : Create a new automation.  The trigger can be Time ( pick a random time ) and in the Action section at the bottom, type Switch into the search field.  You should see options like toggle, turn on, turn off etc.  You then select your template switch as the entity you want to toggle, turn off or turn on.

**Comment by PJH-ZA** [comment]
Thank you for getting back to me.

What do you find is better about the esp32 setup and is it easier to switch
the use_timer on and off?

Thank you

**Reply by hiltctn** [reply]
> Thank you for getting back to me. What do you find is better about the esp32 setup and is it easier to switch the use_timer on and off? Thank you
> […](#)

I had a spare Pi sitting in a drawer and I was going to use it dammit !

Hehe.  Jokes aside, I had a Pi running Venus firmware for a Victron inverter.  When I moved over to Sunsynk, I wanted to reuse it to facilitate local control.  So I started researching and came across KellerZA's software stack first.  While digging into that on various forums, I noticed some folks had cracked the secret to making the Sunsynk work with ESP32's.  So I started looking into that too.  Microcontrollers were new to me at that point,  and I very quickly realised I could make them do whatever I needed them to do ( not relying on code from good people in the interweb )...aaand down that rabbit hole I went.  There are now eight ESP32's ( and two Pi's ) scattered around the house performing various functions. And the one that handles the Sunsynk also runs 3x PWM fans that cool the inverter in the stinking hot garage.

Screenshot of the Timer toggle attached.  As far as HA is concerned, it's just a switch that you use as part of an automation. 

<img width="360" height="555" alt="Screenshot 2025-12-01 at 19 06 47" src="https://github.com/user-attachments/assets/1514f66e-f350-4bd4-92e4-48991c26f887" />

If you feel like a bit of a tinker, ESP32 is the way to go.  If you want something a bit more straightforward, the KellerZA stack is a great option

**Comment by PJH-ZA** [comment]
Excellent. Thank you so much. Definately going to dig into the esp32 stuff
further.

Which esp32 boards did you buy?

Thanks again for getting back to me.

**Reply by hiltctn** [reply]
> Excellent. Thank you so much. Definately going to dig into the esp32 stuff further. Which esp32 boards did you buy? Thanks again for getting back to me.

I went with a full fat ESP32 similar to this [one](https://www.robotics.org.za/ESP32-DEV?search=ESP32) along with [TTL to RS485](https://www.robotics.org.za/RS485-MOD?search=ttl%20to%20rs485) board.  [This](https://www.youtube.com/watch?v=odajXzCnIpE) video explains how to set it up.

[This](https://smarthomeintegrations.co.za/product/smartdeyedonglev4-no-fw-southafrica-only/) is less effort i.e. no soldering iron.  [This](https://powerforum.co.za/topic/25195-a-much-better-way-of-connecting-home-assistant-to-your-deyesunsynk-inverter/#comment-190824) thread explains a lot.

**Comment by PJH-ZA** [comment]
Thank you very much. Definately going to look into these.

Thanks for taking the time to get back to me.

**Reply by hiltctn** [reply]
> Thank you very much. Definately going to look into these. Thanks for taking the time to get back to me.
> […](#)

No worries man.  Good luck.  And have fun !

**Reply by Waz-Cpt** [reply]
@PJH-ZA, The Use Timer switch entity is a native part of the KellerZA sunsynk integration.

What are you missing that you do not have the switch entity for that?

You should see a switch entity in the MQTT sunsynk device. "switch.sunsynk_use_timer".

From there you can integrate that switch into the SlipX sunsyn power flow card and any of your automations you want to create.

**Comment by PJH-ZA** [comment]
Thank you
